src/main/java/picard/util/AdapterMarker.java

```diff
@@ -197,7 +212,14 @@ public AdapterPair adapterTrimIlluminaSingleRead(final SAMRecord read, final int
     public AdapterPair adapterTrimIlluminaPairedReads(final SAMRecord read1, final SAMRecord read2,
                                                              final int minMatchBases, final double maxErrorRate) {
         final AdapterPair ret = ClippingUtility.adapterTrimIlluminaPairedReads(read1, read2, minMatchBases, maxErrorRate, adapters.get());
-        if (ret != null) tallyFoundAdapter(ret);
+        if (ret != null && !thresholdReached) {
+            if(!preAdapterPrunedRecords.containsKey(ret)) {
+                preAdapterPrunedRecords.put(ret, new ArrayList<>());
+            }
+            preAdapterPrunedRecords.get(ret).add(read1);
+            preAdapterPrunedRecords.get(ret).add(read2);
+            tallyFoundAdapter(ret);
+        }
```

Comment:
Is it worth pulling this block into a separate method, to avoid the duplication?